langchain/document_loaders/parsers/audio.py

```diff
@@ -12,10 +12,33 @@ class OpenAIWhisperParser(BaseBlobParser):
     def lazy_parse(self, blob: Blob) -> Iterator[Document]:
         """Lazily parse the blob."""
 
+        import io
         import openai
+        from pydub import AudioSegment
+
+        # Audio file from disk
+        audio = AudioSegment.from_file(blob.path)
+
+        # Define the duration of each chunk in minutes to meet 25MB OpenAI size limit to Whisper API
+        chunk_duration = 20
+
+        # Calculate the chunk duration in milliseconds
+        chunk_duration_ms = chunk_duration * 60 * 1000
+
+        # Split the audio into chunk_duration_ms chunks 
+        for split_number,i in enumerate(range(0, len(audio), chunk_duration_ms)):
+            
+            print(f"Transcribing part {split_number}!")
+
+            # Audio chunk 
+            chunk = audio[i : i + chunk_duration_ms]
+            audio_split = chunk.export(format="mp3").read()
+            file_obj = io.BytesIO(audio_split)
+            file_obj.name = blob.source + f"_part_{split_number}.mp3"
+
+            # Transcribe
+            transcript = openai.Audio.transcribe("whisper-1", file_obj)
 
-        with blob.as_bytes_io() as f:
-            transcript = openai.Audio.transcribe("whisper-1", f)
             yield Document(
```

Comment:
It would be easy to do this. E.g., we can build a single blob from the combined docs:
```
combined_docs = [doc.page_content for doc in docs].join(strings)
```

But, as discussed, it's kind of nice to have the intermediate outputs. 

(The latency is somewhat high - 15 min for 2 hr video.)